### PR TITLE
Correct charset detection

### PR DESCRIPTION
**Pull request by argaen**
Fix for #2197 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bug)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,4 @@ virtualenv.py
 .develop
 .gitconfig
 .flake
+.python-version
```
Comment (argaen) [inline .gitignore:48]: Unrelated with MR. This comes from pyenv

File: tests/test_web_response.py [hunk 2]
```diff
@@ -887,6 +887,14 @@ def test_ctor_charset_without_text():
     assert 'koi8-r' == resp.charset
 
 
+def test_ctor_content_type_with_extra():
```
Comment (argaen) [inline tests/test_web_response.py:890]: Out of curiosity, what does `ctor` stand for?

Comment (fafhrd91) [reply tests/test_web_response.py:890]: constructor

**Comment by asvetlov** [comment]
thanks

**Pull request merged**